### PR TITLE
My fixes to #18624

### PR DESCRIPTION
**Pull request by Hwurzburg**
I am still lost on how terrain following really works and altitude  ref frames,etc.    but I played around with @IamPete1 PR for days until I got it to work.....probably inefficient, but it does work now...added the ability to use therc switch for terrain  disable in Loiter also....had a hard time making things work on every entry into Loiter in every situation...but it does now
SITL tested using the elev stick and changing into and out of Loiter for every case:
-normal mode
-terrain follow in Loiter
-the above with and without alt control

I also felt that we need to maintain current behavior,unless the option bit is set to allow alt control,instead of changing default behavior to have it enabled by default

purposely left each commit unsquashed so that they could be modified easily....as I said, I wacked at it until it worked,not knowinng in depth what I was doing...but I want this functionality for FPV video work

### CODE REVIEW

File: ArduPlane/Attitude.cpp [hunk 1]
```diff
@@ -237,7 +237,12 @@ void Plane::stabilize_stick_mixing_direct()
     aileron = channel_roll->stick_mixing(aileron);
     SRV_Channels::set_output_scaled(SRV_Channel::k_aileron, aileron);
 
-    float elevator = SRV_Channels::get_output_scaled(SRV_Channel::k_elevator);
+    if ((control_mode == &mode_loiter) && (plane.g2.flight_options & FlightOptions::ENABLE_LOITER_ALT_CONTROL)) {
+        // loiter is using altitude control based on the pitch stick, don't use it again here
+        return;
+    }
+
+    int16_t elevator = SRV_Channels::get_output_scaled(SRV_Channel::k_elevator);
```
Comment (tridge) [inline ArduPlane/Attitude.cpp:245]: should be float

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by tridge** [comment]
please adjust comment on update_fbwb_speed_height to include LOITER

File: ArduPlane/Attitude.cpp (hunk 1 above)
Comment (Hwurzburg) [reply ArduPlane/Attitude.cpp:245]: done

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
flight tested on a striver mini quadplane, with and without FLIGHT_OPTIONS bit set
all good!

**Pull request merged**